x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -141,181 +102,73 @@ const AssistantComponent: React.FC<Props> = ({
-  );
-
-  useEffect(() => {
-    if (areConnectorsFetched && conversationsLoaded && Object.keys(conversations).length > 0) {
-      setCurrentConversation((prev) => {
-        const nextConversation =
-          (currentConversationId && conversations[currentConversationId]) ||
-          (isAssistantEnabled &&
-            (conversations[getLastConversationId(conversationTitle)] ||
-              find(conversations, ['title', getLastConversationId(conversationTitle)]))) ||
-          find(conversations, ['title', getLastConversationId(WELCOME_CONVERSATION_TITLE)]);
-
-        if (deepEqual(prev, nextConversation)) return prev;
-
-        const conversationToReturn =
-          (nextConversation &&
-            conversations[
-              nextConversation?.id !== '' ? nextConversation?.id : nextConversation?.title
-            ]) ??
-          conversations[WELCOME_CONVERSATION_TITLE] ??
-          getDefaultConversation({ cTitle: WELCOME_CONVERSATION_TITLE });
-
-        // updated selected system prompt
-        setEditingSystemPromptId(
-          getDefaultSystemPrompt({
-            allSystemPrompts,
-            conversation: conversationToReturn,
-          })?.id
-        );
-        if (
-          prev &&
-          prev.id === conversationToReturn.id &&
-          // if the conversation id has not changed and the previous conversation has more messages
-          // it is because the local conversation has a readable stream running
-          // and it has not yet been persisted to the stored conversation
-          prev.messages.length > conversationToReturn.messages.length
-        ) {
-          return {
-            ...conversationToReturn,
-            messages: prev.messages,
-          };
-        }
-        return conversationToReturn;
-      });
-    }
-  }, [
+  const {
+    currentConversation,
+    currentSystemPromptId,
+    handleCreateConversation,
+    handleOnConversationDeleted,
+    handleOnConversationSelected,
+    handleOnSystemPromptSelectionChange,
+    refetchCurrentConversation,
+    setCurrentConversation,
+    setCurrentSystemPromptId,
+  } = useCurrentConversation({
     allSystemPrompts,
-    areConnectorsFetched,
-    conversationTitle,
     conversations,
-    conversationsLoaded,
-    currentConversationId,
-    getDefaultConversation,
-    getLastConversationId,
-    isAssistantEnabled,
-  ]);
+    defaultConnector,
+    refetchCurrentUserConversations,
+    conversationId: getLastConversationId(conversationTitle),
+    mayUpdateConversations:
+      isFetchedConnectors &&
+      isFetchedCurrentUserConversations &&
+      isFetchedPrompts &&
+      Object.keys(conversations).length > 0,
+  });
 
   // Welcome setup state
-  const isWelcomeSetup = useMemo(() => {
-    // if any conversation has a connector id, we're not in welcome set up
-    return Object.keys(conversations).some(
-      (conversation) => conversations[conversation]?.apiConfig?.connectorId != null
```

Comment:
This was causing a bug, we also need to check that the connector ID is not empty string.

Before:
<img width="1356" alt="welcome-bug" src="https://github.com/user-attachments/assets/9dc96290-e37e-4e79-904f-0655bbdd5ddd">

after: 
<img width="1355" alt="welcome-fix" src="https://github.com/user-attachments/assets/fdb9f99d-d2e0-4c9b-b98c-2b5bd91088c2">